### ISSUE TITLE
UserWarning when computing chunked pint arrays

### ISSUE THREAD

**Issue by TomNicholas** [xarray-contrib/pint-xarray#116]
```python
da = xr.DataArray([1,2,3], dims=['x'], attrs={'units': 'metres'})

chunked = da.pint.quantify().pint.chunk(1)
# chunked2 = da.chunk(1).pint.quantify()  # also happens if I do it in this order instead
```

Everything looks fine here, excellent...

![Screenshot from 2021-07-01 12-31-28](https://user-images.githubusercontent.com/35968931/124158825-66bc4300-da68-11eb-8738-50b4ad4657d3.png)

but when I go to compute then I get a `UserWarning`, even though it returns the correct answer:
```python
chunked.mean().compute()
```
```
/home/tegn500/miniconda3/envs/py38-mamba/lib/python3.8/site-packages/dask/array/core.py:3139: 
UserWarning: Passing an object to dask.array.from_array which is already a Dask collection. This can lead to unexpected behavior.
  warnings.warn(
```

Even if this is working fine then we don't want to be giving warnings to the user ideally.

**Comment by keewis** [comment]
you don't even need the `compute` to get the warning:
```python
In [3]: chunked.mean()
.../lib/python3.8/site-packages/dask/array/core.py:3113: UserWarning: Passing an object to dask.array.from_array which is already a Dask collection. This can lead to unexpected behavior.
  warnings.warn(
Out[3]: 
<xarray.DataArray ()>
dask.array<mean_agg-aggregate, shape=(), dtype=float64, chunksize=(), chunktype=numpy.ndarray>
```
is enough, and computing returns
```
<xarray.DataArray ()>
<Quantity(dask.array<true_divide, shape=(), dtype=float64, chunksize=(), chunktype=numpy.ndarray>, 'meter')>
```
Note that there's no units in the result of `.mean()` and that the return value of `compute` is a `dask` array (wrapped by `pint`) so this is a pretty serious bug (in `xarray`, I think?)

**Comment by TomNicholas** [comment]
Oh dear. Does the other order (`da.chunk(1).pint.quantify()`) behave any differently?

**Comment by keewis** [comment]
no, it doesn't, which is why I believe this is a bug in `xarray`

**Comment by TomNicholas** [comment]
It would be really nice to get this to work before we publish #114 (not that there is any time limit), but I have time now and am keen to help if I can. Should I re-raise this issue on xarray?

**Comment by keewis** [comment]
yes, that would be good.

I didn't test `xarray(pint(dask))` thoroughly, yet, so I expect more to fail. I really hope pydata/xarray#4972 would have caught something like this, which I guess means I should try to finalize that as soon as possible.

**Comment by TomNicholas** [comment]
> Note that there's no units in the result of .mean(), that the return value of compute is a dask array (wrapped by pint) and that we need to compute twice to get the actual result.

Are we definitely seeing the same behaviour as each other? When I do `print(chunked.compute())` (after chunking in either way) I get
```python
<xarray.DataArray (dim_0: 3)>
<Quantity([1 2 3], 'meter')>
Dimensions without coordinates: dim_0
```
which seems right to me?

**Comment by keewis** [comment]
it is correct (which means `.pint.chunk` does not have a bug), but `chunked.mean()` is definitely wrong (I checked both `master` and `v0.2`)

**Comment by keewis** [comment]
with `.compute` I meant that `chunked.mean().compute().compute()` is required to get the result for the mean

**Comment by TomNicholas** [comment]
Right sorry, I had left out the call to mean.
